frontend/src/components/transactions/ReceiptOutcomeRow.tsx

```diff
@@ -0,0 +1,260 @@
+import React from "react";
+import { Row, Col } from "react-bootstrap";
+
+import * as T from "../../libraries/explorer-wamp/transactions";
+import { truncateAccountId } from "../../libraries/formatting";
+import { displayArgs } from "./ActionMessage";
+
+export interface Props {
+  sirReceiptHash: string;
+  receiptsObj: {};
+  receiptsInfoObj: {};
+  senderIsReceiverActions: T.RpcAction[];
+  senderIsReceiver: Boolean;
+}
+
+export default class extends React.Component<Props> {
+  handleStatusInfo = function (receipt: T.ReceiptLocalOutcome) {
+    let statusInfo;
+    if ("SuccessValue" in (receipt.outcome.status as T.ReceiptSuccessValue)) {
+      const { SuccessValue } = receipt.outcome.status as T.ReceiptSuccessValue;
+      if (SuccessValue === null) {
+        statusInfo = "No result";
+      } else if (SuccessValue.length === 0) {
+        statusInfo = "Empty result";
+      } else {
+        statusInfo = (
+          <>
+            <i>Result: </i>
+            {displayArgs(SuccessValue)}
+          </>
+        );
+      }
+    } else if ("Failure" in (receipt.outcome.status as T.ReceiptFailure)) {
+      const { Failure } = receipt.outcome.status as T.ReceiptFailure;
+      statusInfo = (
+        <>
+          <i>Failure: </i>
+          <pre>{JSON.stringify(Failure, null, 2)}</pre>
+        </>
+      );
+    } else if (
+      "SuccessReceiptId" in (receipt.outcome.status as T.ReceiptSuccessId)
+    ) {
+      const { SuccessReceiptId } = receipt.outcome.status as T.ReceiptSuccessId;
+      statusInfo = (
+        <>
+          <i>SuccessReceiptId: </i>
+          <pre>{SuccessReceiptId}</pre>
+        </>
+      );
+    }
+
+    return statusInfo;
+  };
+
+  handleActionInfo = function (
+    receiptObjInfo: T.ReceiptOutcomeInfo,
+    senderIsReceiverActions = []
+  ) {
+    if (
+      receiptObjInfo?.receipt &&
+      (!senderIsReceiverActions || senderIsReceiverActions.length === 0)
+    ) {
+      const { Action: action = null } = receiptObjInfo?.receipt;
+
+      if (action && action.actions && action.actions.length > 0) {
+        const actionsList = action.actions.map((action, index) => (
+          <Row noGutters key={`${action[Object.keys(action)[0]]}_${index}`}>
```

Comment:
Let's extract `Object.keys(action)[0]` into a variable to avoid casting this magic over and over again. I even suggest we preprocess the actions as we do for the [transaction actions](https://github.com/near/near-explorer/blob/35c49e500d6662e70c94d4bab40d3ef72c7ca04d/frontend/src/libraries/explorer-wamp/transactions.ts#L443-L453)

P.S. We should address this weirdness on the JSON API side at some time in the future.